src/lib.rs

```diff
@@ -1724,12 +1755,7 @@ impl<'a> Iterator for ChunkIter<'a> {
         unsafe {
             let foot = self.footer?;
             let foot = foot.as_ref();
-            let data = foot.data.as_ptr() as usize;
-            let ptr = foot.ptr.get().as_ptr() as usize;
-            debug_assert!(data <= ptr);
-            debug_assert!(ptr <= foot as *const _ as usize);
-
-            let len = foot as *const _ as usize - ptr;
+            let (ptr, len) = foot.as_raw_parts();
```

Comment:
`ChunkIter` could contain a `raw: ChunkRawIter` inside of it and then this impl could just be

```rust
let (ptr, len) = self.raw.next()?;
let slice = slice::from_raw_parts(ptr as *const mem::MaybeUninit<u8>, len);
Some(slice)
```

instead of duplicating the footer-walking logic.